### DISCUSSION TITLE
ImportError: cannot import name 'GeometryCollection' from 'shapely' (/opt/conda/lib/python3.10/site-packages/shapely/__init__.py)

### DISCUSSION THREAD

**Discussion by atutuncuogluu** [obss/sahi#1061] [Q&A] [answered]
If I install roboflow with pip first, I encounter this situation.

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[47], line 1
----> 1 from sahi.slicing import slice_coco
      2 from sahi.utils.file import load_json
      4 from PIL import Image, ImageDraw

File /opt/conda/lib/python3.10/site-packages/sahi/__init__.py:3
      1 __version__ = "0.11.18"
----> 3 from sahi.annotation import BoundingBox, Category, Mask
      4 from sahi.auto_model import AutoDetectionModel
      5 from sahi.models.base import DetectionModel

File /opt/conda/lib/python3.10/site-packages/sahi/annotation.py:9
      5 from typing import Dict, List, Optional
      7 import numpy as np
----> 9 from sahi.utils.coco import CocoAnnotation, CocoPrediction
     10 from sahi.utils.cv import (
     11     get_bbox_from_coco_segmentation,
     12     get_bool_mask_from_coco_segmentation,
     13     get_coco_segmentation_from_bool_mask,
     14 )
     15 from sahi.utils.shapely import ShapelyAnnotation

File /opt/conda/lib/python3.10/site-packages/sahi/utils/coco.py:17
     14 from typing import Dict, List, Optional, Set, Union
     16 import numpy as np
---> 17 from shapely import GeometryCollection, MultiPolygon, Polygon
     18 from shapely.validation import make_valid
     19 from tqdm import tqdm

ImportError: cannot import name 'GeometryCollection' from 'shapely' (/opt/conda/lib/python3.10/site-packages/shapely/__init__.py)

# from https://github.com/shapely/shapely current shapely/__init__py :

from shapely.lib import GEOSException  # NOQA
from shapely.lib import Geometry  # NOQA
from shapely.lib import geos_version, geos_version_string  # NOQA
from shapely.lib import geos_capi_version, geos_capi_version_string  # NOQA
from shapely.errors import setup_signal_checks  # NOQA
from shapely._geometry import *  # NOQA
from shapely.creation import *  # NOQA
from shapely.constructive import *  # NOQA
from shapely.predicates import *  # NOQA
from shapely.measurement import *  # NOQA
from shapely.set_operations import *  # NOQA
from shapely.linear import *  # NOQA
from shapely.coordinates import *  # NOQA
from shapely.strtree import *  # NOQA
from shapely.io import *  # NOQA

from shapely.geometry import (  # NOQA
    Point,
    LineString,
    Polygon,
    MultiPoint,
    MultiLineString,
    MultiPolygon,
    GeometryCollection,
    LinearRing,
)

from shapely import _version

__version__ = _version.get_versions()["version"]

setup_signal_checks()

**Comment by fcakyon** [comment]
Hey @atutuncuogluu. Are you sure your shapely version is shapely>=1.8.0?

**Reply by atutuncuogluu** [reply, marked as answer]
When I installed some libraries first, I was getting an error because pip made a version correction. When I changed the order of installing the libraries, my problem was solved.

> Hey @atutuncuogluu. Are you sure your shapely version is shapely>=1.8.0?

**Comment by nghible** [comment]
Hi, I also encountered this error using shapely 1.8.2. Shapely 1.8.2 does not import GeometryCollection in the __init__.py file here: https://github.com/shapely/shapely/blob/1.8.2/shapely/__init__.py

But shapely 2.0 does: https://github.com/shapely/shapely/blob/2.0.0/shapely/__init__.py

So after updating to 2.0, it's fine for me. But you guys should fix the requirements.txt.

**Reply by weiji14** [reply]
Opened a PR to pin to `shapely>2.0.0` at https://github.com/obss/sahi/pull/1101
